sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -1890,17 +1887,21 @@ public void populateDisplayData(DisplayData.Builder builder) {
     private final DisplayData.ItemSpec<? extends Class<?>> fnDisplayData;
     private final SerializableFunction<? super K, Integer> hotKeyFanout;
 
-    private PerKeyWithHotKeyFanout(String name,
+    private PerKeyWithHotKeyFanout(
         PerKeyCombineFn<? super K, ? super InputT, ?, OutputT> fn,
         DisplayData.ItemSpec<? extends Class<?>> fnDisplayData,
         SerializableFunction<? super K, Integer> hotKeyFanout) {
-      super(name);
       this.fn = fn;
       this.fnDisplayData = fnDisplayData;
       this.hotKeyFanout = hotKeyFanout;
     }
 
     @Override
+    protected String getKindString() {
+      return String.format("Combine.perKeyWithFanout(%s)", NameUtils.approximateSimpleName(fn));
```

Comment:
I think in the past it would just be something like `Largest.PerKey` because (1) each Combine transform defined its own name and (2) the "with hot key fanout" just delegated to the name it was passed.

In this change, I have converted the Combine transforms to be more like `Combine.perKey(Top(Largest))` so the name indicates that this is a per-key combine using the `Top` combine fn with the "largest" comparator.

Since this change also enabled indicating that this is using hot key fanout, I did so -- but I don't think that is necessary. I'm happy to drop this or change the name back to `Combine.perKey(...)`.

I don't think we have any promise that the name is a valid expression -- in the past `Largest.PerKey` was not valid, nor is `Combine.perKey(Top(Largest))`...